src/arm/cpu.rs

```diff
@@ -0,0 +1,30 @@
+use cfg_if::cfg_if;
+use libc;
+
+pub const DAV1D_ARM_CPU_FLAG_NEON: libc::c_uint = 1 << 0;
+pub const NEON_HWCAP: libc::c_ulong = 1 << 12;
+
+#[no_mangle]
+#[cold]
+pub unsafe extern "C" fn dav1d_get_cpu_flags_arm() -> libc::c_uint {
+    let mut flags = 0;
+
+    cfg_if! {
+        if #[cfg(any(
+            target_arch = "aarch64",
```

Comment:
This is `aarch64 || windows || macos`. Do you want something like `aarch64 && (windows || macos)`?

Reply:
The original C was

```c
#if defined(__ARM_NEON) || defined(__APPLE__) || defined(_WIN32) || ARCH_AARCH64
```

So that's what I was trying to replicate 🤷

Reply:
The caller of this function already ensures that the arch is `arm` or `aarch64` which why the condition in C works. The question, as I understand it, is whether we want something close to the logic of the original code (caller ensures we're on ARM) or something that is more robust (e.g. the logic that Andrei suggested).

Reply:
Oh I didn't realize in my original comment that this also applied to 32-bit ARM. The conditional makes more sense now: 64-bit or Windows or Apple ARM.